### PR TITLE
Implement prettier precommit hook

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

- Adds a git precommit hook to run prettier on staged files

### How to review 

- `yarn`
- open a file and remove formatting that would be applied with prettier (eg. semicolons)
- stage and commit file/s
- check that prettier reformats staged code

### CODE REVIEW

**Comment by hamishtaplin** [comment]
@samiwel It only prettifies the files that have been staged, thankfully!

**Comment by hamishtaplin** [comment]
[Enable Prettier (3)](https://trello.com/c/bIYYAN8F/27-enable-prettier-3)

File: package.json [hunk 1]
```diff
@@ -10,7 +10,14 @@
...
     "deploy": "npm run build&&gh-pages -d build",
     "storybook": "export NODE_PATH=src/ && start-storybook -p 9009 -s public",
-    "build-storybook": "build-storybook -s public"
+    "build-storybook": "build-storybook -s public",
+    "precommit": "lint-staged"
+  },
+  "lint-staged": {
+    "*.js": [
+      "prettier --write",
+      "git add"
```
Comment (collisdigital) [inline package.json]: risk this could add stuff you don't want as well?

Comment (hamishtaplin) [inline package.json]: it only runs against files that are already staged afaik:

https://github.com/okonet/lint-staged

Comment (WickyNilliams) [inline package.json]: Reading the source it seems that way, yeah

**Comment by WickyNilliams** [comment]
I used https://github.com/prettier/eslint-config-prettier to determine which eslint rules would clash with prettier's formatting. This is the output:

```
The following rules are unnecessary or might conflict with Prettier:

- comma-spacing
- dot-location
- eol-last
- indent
- jsx-quotes
- new-parens
- no-extra-semi
- no-mixed-spaces-and-tabs
- no-multi-spaces
- no-multiple-empty-lines
- no-trailing-spaces
- no-whitespace-before-property
- object-curly-spacing
- react/jsx-closing-bracket-location
- react/jsx-curly-spacing
- react/jsx-equals-spacing
- react/jsx-indent
- react/jsx-indent-props
- react/jsx-max-props-per-line
- react/jsx-space-before-closing
- react/jsx-wrap-multilines
- rest-spread-spacing
- semi
- semi-spacing
- space-infix-ops
- unicode-bom

The following rules are enabled with options that might conflict with Prettier. See:
https://github.com/prettier/eslint-config-prettier#special-rules

- quotes

The following rules are enabled but cannot be automatically checked. See:
https://github.com/prettier/eslint-config-prettier#special-rules

- no-mixed-operators
```

They offer a prettier-specific set of eslint rules which will disable any rules that overlap with prettier . We should add this to our `.eslintrc`

**Pull request merged**